blosc/blosc.c

```diff
@@ -1355,18 +1364,21 @@ int blosc_run_decompression_with_context(struct blosc_context* context,
 
   /* Read the header block */
   version = context->src[0];                        /* blosc format version */
-  versionlz = context->src[1];                      /* blosclz format version */
+  context->versionlz = context->src[1];             /* blosclz format version */
 
   context->header_flags = (uint8_t*)(context->src + 2);           /* flags */
   context->typesize = (int32_t)context->src[3];      /* typesize */
   context->sourcesize = sw32_(context->src + 4);     /* buffer size */
   context->blocksize = sw32_(context->src + 8);      /* block size */
-  ctbytes = sw32_(context->src + 12);               /* compressed buffer size */
 
-  /* Unused values */
-  version += 0;                             /* shut up compiler warning */
-  versionlz += 0;                           /* shut up compiler warning */
-  ctbytes += 0;                             /* shut up compiler warning */
+  if (version != BLOSC_VERSION_FORMAT) {
+    /* Version from future */
```

Comment:
May it be worth to print some informative message before giving up?  Something like:

```
printf("Unrecognized header version %d\n", version);
```

Reply:
I don't like libraries that print messages, that descision should be given to the caller. Before we have a way to return error messages, I prefer silent errors. Maybe we can return -10 and document what it means in the API documentation?

Reply:
I see.  However, the current approach is to print an informative message before returning a negative number.  I don't think that changing this behavior in this PR in just a couple of places is a good idea.  Probably the good thing to do is to detect all the places where a negative code is returned, make sure that there are no repetitions and document these codes.  One can even think on creating a new API (`blosc_print_error(int)?) for more verbose output.  But this would be another PR.

Reply:
Yes, we need at least to move the error printing out of the helper threads to avoid printing garbled output. I can do that refactor in a separate pull request with the existing error messages, and push that first, if you want.

Reply:
Well, maybe a printf is not strictly necessary.  Perhaps a `char* blosc_get_error(int)` would be enough.  A better approach would be to define an enumerated for the different errors and some functions for returning proper messages.  The [Zstandard approach](https://github.com/facebook/zstd/blob/dev/lib/common/zstd_common.c#L31) could be a good example to follow.

Reply:
That sounds like a good idea to me, the Zstd approach resembles what I've seen in other libraries. I also don't think a library should print things unless told to - what to do with the error should be the user's decision (where the user here is a programmer).